server/mdm/apple/apple_mdm.go

```diff
@@ -47,3 +60,120 @@ func resolveURL(serverURL, relPath string) (string, error) {
 	u.Path = path.Join(u.Path, relPath)
 	return u.String(), nil
 }
+
+type DEPSyncer struct {
+	depStorage nanodep_storage.AllStorage
+	syncer     *depsync.Syncer
+	logger     kitlog.Logger
+}
+
+func (d *DEPSyncer) Run(ctx context.Context) error {
+	profileUUID, profileModTime, err := d.depStorage.RetrieveAssignerProfile(ctx, DEPName)
+	if err != nil {
+		return err
+	}
+	if profileUUID == "" {
+		d.logger.Log("msg", "DEP profile not set, nothing to do")
+		return nil
+	}
+	cursor, cursorModTime, err := d.depStorage.RetrieveCursor(ctx, DEPName)
+	if err != nil {
+		return err
+	}
+	// If the DEP Profile was changed since last sync then we clear
+	// the cursor and perform a full sync of all devices and profile assigning.
+	if cursor != "" && profileModTime.After(cursorModTime) {
+		d.logger.Log("msg", "clearing device syncer cursor")
+		if err := d.depStorage.StoreCursor(ctx, DEPName, ""); err != nil {
+			return err
+		}
+	}
+	return d.syncer.Run(ctx)
+}
+
+func NewDEPSyncer(
+	ds fleet.Datastore,
+	depStorage nanodep_storage.AllStorage,
+	logger kitlog.Logger,
+	loggingDebug bool,
+) *DEPSyncer {
+	depClient := NewDEPClient(depStorage, ds, logger)
+	assignerOpts := []depsync.AssignerOption{
+		depsync.WithAssignerLogger(logging.NewNanoDEPLogger(kitlog.With(logger, "component", "nanodep-assigner"))),
+	}
+	if loggingDebug {
+		assignerOpts = append(assignerOpts, depsync.WithDebug())
+	}
+	assigner := depsync.NewAssigner(
+		depClient,
+		DEPName,
+		depStorage,
+		assignerOpts...,
+	)
+
+	syncer := depsync.NewSyncer(
+		depClient,
+		DEPName,
+		depStorage,
+		depsync.WithLogger(logging.NewNanoDEPLogger(kitlog.With(logger, "component", "nanodep-syncer"))),
+		depsync.WithCallback(func(ctx context.Context, isFetch bool, resp *godep.DeviceResponse) error {
+			n, err := ds.IngestMDMAppleDevicesFromDEPSync(ctx, resp.Devices)
+			switch {
+			case err != nil:
+				level.Error(kitlog.With(logger)).Log("err", err)
+				sentry.CaptureException(err)
+			case n > 0:
+				level.Info(kitlog.With(logger)).Log("msg", fmt.Sprintf("added %d new mdm device(s) to pending hosts", n))
+			default:
+				// ok
```

Comment:
Nit but this default clause should be removed? Though it seems like you're not alone in liking this style, I don't want to be a style bully :D it's just that I find it weird because we wouldn't typically do this for an `if else` statement? But anyway, obviously no big deal either way.

Reply:
In this case this is code that I moved, but in other cases I have added an empty `default` clause just because the nearby coded also did it lol

your argument about `if / else` makes sense and made me chuckle:


```go
if thing {
  doSomething()
} else {
  // ok
}
```

I'm with you about not having strong feelings about this but I removed it!